src/libstore/build/local-derivation-goal.cc

```diff
@@ -354,9 +354,19 @@ static void preloadNSS() {
     std::call_once(dns_resolve_flag, []() {
         struct addrinfo *res = NULL;
 
-        if (getaddrinfo("this.pre-initializes.the.dns.resolvers.invalid.", "http", NULL, &res) != 0) {
+        /* nss will only force the "local" (not through nscd) dns resolution if its on the LOCALDOMAIN.
+           We need the resolution to be done locally, as nscd socket will not be accessible in the
+           sandbox. */
+        char * previous_env = getenv("LOCALDOMAIN");
+        setenv("LOCALDOMAIN", "invalid", 1);
```

Comment:
Changing the environment is not thread-safe (and there *can* be other threads running concurrently, like the auto-GC thread).

Reply:
I completely agree with that, but I haven't found any other way to force libc to load nss components.

What would you think about moving the call to preloadNSS at the start of the nix-daemon? There would only be on thread and no thread-safety issue.

Reply:
One alternative is to inject glibc inside the fetcher sandbox. I don't believe this would cause any major non-reproducibility.
I can work on that, let me know what you think is best.

Reply:
Doing it at the start in `initNix()` sounds good to me.